garrysmod/lua/includes/extensions/player.lua

```diff
@@ -241,6 +241,17 @@ end
 
 end
 
+--[[---------------------------------------------------------
+	OnLadder
+	Returns true if the player is climbing a ladder
+-----------------------------------------------------------]]
+
+function meta:OnLadder()

```

Comment:
```suggestion
function meta:IsOnLadder()
```
`On` prefixes should be reserved for Outputs, besides, we already have a few `IsOn` functions.
There's also to consider that NextBot can also use ladders, making it not bound specifically to the player.

If you want to pollute this PR with unrelated changes...